File: frontend/src/objects/order/core/OrderTimelineEntry.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type { DisplayText, IsoDateTime, OrderStatus } from '@/objects/core/SharedObjects'

export type OrderTimelineEntry = {
  status: OrderStatus
  note: DisplayText
  at: IsoDateTime
}

File: backend/src/main/scala/services/order/objects/OrderTimelineEntry.scala
package services.order.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*
import system.objects.*

final case class OrderTimelineEntry(status: OrderStatus, note: DisplayText, at: IsoDateTime)
object OrderTimelineEntry:
  given Encoder[OrderTimelineEntry] = deriveEncoder
  given Decoder[OrderTimelineEntry] = deriveDecoder
